content/knowledge/scalinghistory.md

```diff
@@ -0,0 +1,135 @@
+
+The concept of Blockchain or Cryptography are not new and have been around before the 2000's.
+Still there was no concept of a truly digital currency/value due to the double spending problem.
+For example, say I have $100 worth of digital value and I send $100 to both Alice and Bob at the same time.
+There used to be no fully robust way to stop this from happening.
+
+## 1. Proof of Work (PoW)
+
+   Then came Satoshi Nakamoto who introduced the concept of Proof of Work (PoW) on top of a Blockchain which acts as a distributed ledger of all the transactions of value/currency.
+   So now imagine I have 100 Bitcoins.
+   There are many machines/nodes which have this information that I have 100 Bitcoins.
+   If I send 50 of them to Alice then certain things happen before it gets "finalised".
+   These things include me sending out this transaction info, some number of machines/nodes accepting it (along with transactions from others), verifying that I am truly the one sending it out and then they solve a complicated problem which takes a lot of computing power.
+   The first one to solve it actually gets to "write" these transactions in the form of a block and that's how it is finalised.
+   A good interactive visualisation of how PoW consensus might work and create a chain of blocks can be found [here](https://youtu.be/_160oMzblY8).
+
+PoW by itself isn’t very new.
+Original similar implementation was thought of in 1992 to [combat email spam](https://en.wikipedia.org/wiki/Hashcash).
+So every email will have some form of simple proof of work so that it is easy for me to send email to a friend but gets very hard for a spammer to send millions of emails at the same time.
+
+In theory, PoW can scale infinitely.
+But with it comes the extra overhead of computational complexity which may not be practical in a truly global sense.
+Also some PoW based blockchains like Ethereum1.0 suffer from huge transaction fees as well as low throughput of 10-15 transactions per second which make them impractical for true global usage.
+   Therefore the PoW consensus based blockchains can achieve decentralisation and security but not scalability.
+
+## 2. Federated Byzantine Agreement (FBA)
+
+   After Bitcoin and PoW came into picture in 2008, another class of consensus protocols surfaced around 2014 which tried to tackle a different problem known as “Byzantine Generals Problem”.
+   This is the problem when there are various nodes communicating with each other and a few of them are “Byzantine” which means either faulty or malicious.
+   A good consensus protocol should be able to tackle this issue in a decentralised manner for better security of the network.
+   One of the consensus algorithms that manage to do it is called “Federated Byzantine Agreement”.
+   This was introduced by Stellar.
```

Comment:
Real FBA was indeed first defined, proven correct and implemented by Stellar. However, Stellar was based on the ideas of Ripple, with a more theory-based approach.